runtime/lua/vim/lsp/protocol.lua

```diff
@@ -535,6 +535,9 @@ function protocol.make_client_capabilities()
       colorProvider = {
         dynamicRegistration = true,
       },
+      linkedEditingRange = {
+        dynamicRegistration = false,
```

Comment:
IMO, this can be true. because this implementaion is checking capabilities on-demand in https://github.com/neovim/neovim/pull/34388/files#diff-14652e7f4fb9a2b4871113310a3e66c5ca2bd1646457e590201530d1e790f6b0R150.

Reply:
I would lean towards not announcing dynamic registration support for now, even if it technically would work. As @ribru17 mentioned this might require keeping some extra state which we can add in a follow-up.